### PR TITLE
Make the UI work again

### LINKED ISSUE #1893: Make the UI work again
Using the new API version
Don't mind panel surveys

For #1886
Depends on #1892

### PR DESCRIPTION
**Pull request by aeatencio**
Without taking the panel surveys into account (they are working partially) the UI seems to be working just fine again.

Known pending issues (they will be fixed in other PR, for #1887):
- Add `generates_panel_survey` property to surveys.
- On launch, create the panel survey when a survey with the `generates_panel_survey` flag is ON.
- Name survey occurrences `YYYY-MM-DD`.
- Add calculated `is_repeatable` property to panel_surveys.
- Finish the new `PanelSurveyShow` component. It includes:
  - Reuse part of the existing `FolderShow` component.
  - Add the `new occurrence` button.
  - Fix the breadcrumb to show the panel survey name from the panel survey occurrence page.
- On panel survey creation, make the model validate that an existing survey is necessarily assigned to the new instance. This is to assure a panel survey is never empty (with no occurrences).
- On the panel survey model, forbid changing `panel_survey_id` to `nil` and only allow changing it when the survey is ready to lunch and `generates_panel_survey` is ON.
- Be aware of the current UI will take the panel survey name from the survey that generates it. After that, the end-user won't be able of renaming the panel survey.

Fix #1893

### CODE REVIEW

**Review by anaPerezGhiglia — APPROVED** [review]
Looking good!

File: web/controllers/survey_controller.ex [hunk 1]
```diff
@@ -273,12 +273,15 @@ defmodule Ask.SurveyController do
...
-      "running" ->
-        send_resp(conn, :bad_request, "")
+    # TODO: We should move this validations to SurveyAction in the future
+    cond do
+      survey.state == "running" ->
+        send_resp(conn, :bad_request, "Cannot delete a running survey")
 
-      _ ->
+      only_survey_in_panel_survey?(survey) ->
+        send_resp(conn, :forbidden, "Cannot delete the only survey of a panel survey")
```
[the remark this answers is not in the document]
Comment (aeatencio) [reply web/controllers/survey_controller.ex:282]: Good catch! Thanks!!

File: web/static/js/components/surveys/PanelSurveyShow.jsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        if (value[surveyId].state != 'not_ready') {
+          dispatch(respondentActions.fetchRespondentsStats(projectId, surveyId))
+        }
+      }
+    })
+    dispatch(folderActions.fetchFolders(projectId))
+    dispatch(panelSurveysActions.fetchPanelSurveys(projectId))
+    if (!panelSurvey) {
+      dispatch(panelSurveyActions.fetchPanelSurvey(projectId, panelSurveyId))
+    }
```
[the remark this answers is not in the document]
Comment (aeatencio) [reply web/static/js/components/surveys/PanelSurveyShow.jsx:58]: Good questions! But... I'm not sure we want to know the answers! =P

**Comment by aeatencio** [comment]
@anaPerezGhiglia, let's fix everything regarding the `PanelSurveyShow` component in the next PRs. Otherwise, we'll have several problems with rebasing the following branches.

**Pull request merged**